gatekeeper/http/http.go

```diff
@@ -28,11 +31,35 @@ type Gatekeeper struct {
 }
 
 // NewGatekeeper returns a new Gatekeeper.
-func NewGatekeeper(org, team string, cfg *config.Config, api *api.Client) *Gatekeeper {
+func NewGatekeeper(org, team, certpath, keypath string, cfg *config.Config, api *api.Client) (*Gatekeeper, error) {
 	server := &http.Server{
 		Addr: cfg.GatekeeperAddress,
 	}
 
+	if certpath == "" || keypath == "" {
+		log.Printf("Starting Gatekeeper without SSL")
+	} else {
+		keypair, err := tlsKeypair(certpath, keypath)
+		if err != nil {
+			return nil, err
+		}
+
+		tlsConfig := &tls.Config{
+			MinVersion:               tls.VersionTLS12,
+			CurvePreferences:         []tls.CurveID{tls.CurveP521, tls.CurveP384, tls.CurveP256},
```

Comment:
Have we checked the right configuration for these curves (e.g. are they the widely recommended ones)? If I remember correclty these curves have been backdoored by the NSA (since they're NIST curves).